python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -74,16 +74,22 @@ def __setattr__(self, attr, value):
         if attr == "namePattern":
             actual_value = os.path.basename(value)
             this_localDir = os.path.dirname(value)
-            super(MassStorageFile, self).__setattr__('localDir', this_localDir)
+            if this_localDir:
+                super(MassStorageFile, self).__setattr__('localDir', this_localDir)
```

Comment:
Is there a reason you're circumventing the current class' implementation of `__setattr__` here?